server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -376,5 +439,85 @@ AnalysisStats analysisStats() {
         ClusterSnapshotStats clusterSnapshotStats() {
             return clusterSnapshotStats;
         }
+
+        public Map<String, RemoteClusterStats> getRemoteStats() {
+            return remoteStats;
+        }
+    }
+
+    private static boolean doRemotes(ClusterStatsRequest request) {
+        return CCS_TELEMETRY_FEATURE_FLAG.isEnabled() && request.doRemotes();
+    }
+
+    private class RemoteStatsFanout extends CancellableFanOut<String, RemoteClusterStatsResponse, Map<String, RemoteClusterStats>> {
+        private final Map<String, RemoteClusterStatsResponse> responses = new ConcurrentHashMap<>();
+        private final Executor requestExecutor;
+        private final Task task;
+        private final TaskId taskId;
+        private final Collection<String> remotes;
+
+        RemoteStatsFanout(Task task, Executor requestExecutor, Collection<String> remotes) {
+            this.task = task;
+            this.requestExecutor = requestExecutor;
+            this.taskId = new TaskId(clusterService.getNodeName(), task.getId());
+            this.remotes = remotes;
+        }
+
+        @Override
+        protected void sendItemRequest(String clusterAlias, ActionListener<RemoteClusterStatsResponse> listener) {
+            var remoteRequest = new RemoteClusterStatsRequest();
+            var remoteClusterClient = remoteClusterService.getRemoteClusterClient(
+                clusterAlias,
+                requestExecutor,
+                RemoteClusterService.DisconnectedStrategy.RECONNECT_IF_DISCONNECTED
+            );
+            remoteRequest.setParentTask(taskId);
+            remoteClusterClient.execute(TransportRemoteClusterStatsAction.REMOTE_TYPE, remoteRequest, listener);
+        }
+
+        @Override
+        protected void onItemResponse(String clusterAlias, RemoteClusterStatsResponse response) {
+            if (isCancelled() == false) {
+                responses.put(clusterAlias, response);
+            }
+        }
+
+        @Override
+        protected void onItemFailure(String clusterAlias, Exception e) {
+            logger.warn("Failed to get remote cluster stats for [{}]: {}", clusterAlias, e);
+        }
+
+        private boolean isCancelled() {
```

Comment:
I think `CancellableFanOut` already does the right thing with cancellation, no need to check it yourself too?